### PR TITLE
Made starting platform

### PR DESCRIPTION
**Pull request by josepharhar**
#96

### CODE REVIEW

File: src/game_state/GameState.cpp [hunk 1]
```diff
@@ -30,15 +38,24 @@ std::shared_ptr<Player> GameState::GetPlayer() {
...
 
-uint64_t GameState::GetTimingStartTick() {
-  return timing_start_tick;
+uint64_t GameState::GetStartTick() {
+  return start_tick;
+}
+
+uint64_t GameState::GetMusicStartTick() {
+  // the music start tick is always dependent on the start tick
+  return start_tick + PREGAME_SECONDS * TICKS_PER_SECOND;
```
Comment (bbeck13) [inline src/game_state/GameState.cpp:54]: Yea I'm going to the party latter but first I have to `PREGAME_SECONDS`

File: src/game_updater/GameUpdater.cpp [hunk 2]
```diff
@@ -58,11 +58,20 @@ void GameUpdater::Init(std::shared_ptr<GameState> game_state) {
 }
 
 void GameUpdater::Update(std::shared_ptr<GameState> game_state) {
+  std::shared_ptr<sf::Music> music = game_state->GetLevel()->getMusic();
+
   // check to see if the music ended, in which case the level is complete
-  if (game_state->GetLevel()->getMusic()->getStatus() ==
-      sf::Music::Status::Stopped) {
+  if (music->getStatus() == sf::Music::Status::Stopped &&
+      music->getPlayingOffset() != sf::Time::Zero) {
```
Comment (bbeck13) [inline src/game_updater/GameUpdater.cpp]: This looks a little confusing may I suggest
`((music->getStatus() == sf::Music::Status::Stopped) && (music->getPlayingOffset() != sf::Time::Zero))`
A little extra parenthesis never hurt

File: src/generator/LevelGenerator.cpp [hunk 3]
```diff
@@ -88,7 +88,12 @@ std::shared_ptr<Level> LevelGenerator::generateLevel() {
   double xPos = -1, yPos = 2, zPos = -5, power = 0, lastPower = 0;
   int lastSample = samplesPerPlatform, ups = 0, downs = 0, wobble = 0,
       dropping = 0, moving = 0;
-  objs->push_back(std::make_shared<Platform>(glm::vec3(xPos, yPos, zPos)));
+
+  double pregame_platform_width = DELTA_X_PER_SECOND * PREGAME_SECONDS;
```
Comment (bbeck13) [inline src/generator/LevelGenerator.cpp]: I get turnt on that `pregame_platform_width`

File: src/helpers/TimingConstants.h [hunk 4]
```diff
@@ -31,4 +31,7 @@
 #define PLAYER_JUMP_VELOCITY PLAYER_JUMP_DELTA_Y_PER_TICK
 #define PLAYER_GRAVITY (PLAYER_JUMP_DELTA_Y_PER_TICK / 40.0)  // dy^2 per tick
 
+// Amount of seconds before the music starts
+#define PREGAME_SECONDS 1
```
Comment (bbeck13) [inline src/helpers/TimingConstants.h:35]: 1 second is all it takes

**Review by bbeck13 — COMMENTED** [review]
Looks good I want to push in https://github.com/bbeck13/RhythmRunner/pull/101 before if that's ok there shouldn't be any conflicts

File: src/game_state/GameState.cpp (hunk 1 above)
Comment (josepharhar) [reply src/game_state/GameState.cpp:54]: 👌

File: src/game_updater/GameUpdater.cpp (hunk 2 above)
Comment (josepharhar) [reply src/game_updater/GameUpdater.cpp]: 👍

**Comment by josepharhar** [comment]
Rebased and updated for comments

**Comment by bbeck13** [comment]
CR 🏷 QA 👍

**Pull request merged**